Sources/MessagingInApp/State/Core/Store.swift

```diff
@@ -0,0 +1,188 @@
+        transformedSubscription: Subscription<SelectedState>?
+    )
+        where S.StoreSubscriberStateType == SelectedState {
+        let subscriptionBox = self.subscriptionBox(
+            originalSubscription: originalSubscription,
+            transformedSubscription: transformedSubscription,
+            subscriber: subscriber
+        )
+
+        subscriptions.update(with: subscriptionBox)
+
+        if let state = state {
+            originalSubscription.newValues(oldState: nil, newState: state)
+        }
+    }
+
+    open func subscribe<S: StoreSubscriber>(_ subscriber: S)
+        where S.StoreSubscriberStateType == State {
+        guard subscriptionsAutomaticallySkipRepeats else {
+            subscribe(subscriber, transform: nil)
+            return
+        }
+        subscribe(subscriber, transform: { $0.skipRepeats() })
+    }
+
+    func subscriptionBox<T>(
+        originalSubscription: Subscription<State>,
+        transformedSubscription: Subscription<T>?,
+        subscriber: AnyStoreSubscriber
+    ) -> SubscriptionBox<State> {
+        SubscriptionBox(
+            originalSubscription: originalSubscription,
+            transformedSubscription: transformedSubscription,
+            subscriber: subscriber
+        )
+    }
+
+    open func unsubscribe(_ subscriber: AnyStoreSubscriber) {
+        #if swift(>=5.0)
+        if let index = subscriptions.firstIndex(where: { $0.subscriber === subscriber }) {
+            subscriptions.remove(at: index)
+        }
+        #else
+        if let index = subscriptions.index(where: { $0.subscriber === subscriber }) {
+            subscriptions.remove(at: index)
+        }
+        #endif
+    }
+
+    // swiftlint:disable:next identifier_name
+    open func _defaultDispatch(action: InAppMessageAction) {
+        guard !isDispatching.value else {
+            raiseFatalError(
+                "ReSwift:ConcurrentMutationError- Action has been dispatched while" +
+                    " a previous action is being processed. A reducer" +
+                    " is dispatching an action, or ReSwift is used in a concurrent context" +
+                    " (e.g. from multiple threads). Action: \(action)"
+            )
+        }
+
+        isDispatching.value { $0 = true }
+        let newState = reducer(action, state)
+        isDispatching.value { $0 = false }
+
+        state = newState
+    }
+
+    open func dispatch(_ action: InAppMessageAction) {
+        dispatchFunction(action)
+    }
+}
+
+// MARK: Skip Repeats for Equatable States
+
+extension Store {
+    public func subscribe<SelectedState: Equatable, S: StoreSubscriber>(
+        _ subscriber: S, transform: ((Subscription<State>) -> Subscription<SelectedState>)?
+    ) where S.StoreSubscriberStateType == SelectedState {
+        let originalSubscription = Subscription<State>()
+
+        var transformedSubscription = transform?(originalSubscription)
+        if subscriptionsAutomaticallySkipRepeats {
```

Comment:
`subscriptionsAutomaticallySkipRepeats` is always true, i think we can remove it.

Reply:
Yup, removed